assets/js/components/Dashboard/DatePicker.tsx

```diff
@@ -0,0 +1,78 @@
+import moment from "moment";
+import React, { ComponentType, useRef, useState } from "react";
+import { Form, Overlay, Popover } from "react-bootstrap";
+import Calendar from "react-calendar";
+
+interface DatePickerProps {
+  selectedDate: string;
+  onChange: (date: string) => void;
+  minDateString?: string;
+  maxDateString?: string;
+}
+
+const DatePicker: ComponentType<DatePickerProps> = ({
+  selectedDate,
+  onChange,
+  minDateString,
+  maxDateString,
+}: DatePickerProps) => {
+  const [inputString, setInputString] = useState(selectedDate);
+  const [showOverlay, setShowOverlay] = useState(false);
+  const ref = useRef(null);
+  const minDate = minDateString ? new Date(minDateString) : undefined;
+  const maxDate = maxDateString ? new Date(maxDateString) : undefined;
+
+  return (
+    <>
+      <Form
+        ref={ref}
+        className="date-picker"
+        onSubmit={(event) => {
+          event.preventDefault();
+          onChange(inputString);
+        }}
+        onClick={() => setShowOverlay(!showOverlay)}
+      >
+        <Form.Control
+          value={inputString}
+          onChange={(input) => {
+            const inputValue = input.target.value;
+            setInputString(inputValue);
+            onChange(inputValue);
```

Comment:
**question:** is there a reason to maintain this state locally instead of having this be a fully [controlled component](https://react.dev/learn/sharing-state-between-components#controlled-and-uncontrolled-components)

Reply:
Hmm, definitely not. Will adjust.